middlewares/userValidations.js

```diff
@@ -109,6 +109,69 @@ export default {
       return next(error);
     }
     next();
+  },
+  loginParamsValidator(req, res, next) {
+    const availableParams = ['usernameOrEmail', 'password'];
+
+    const errorArray = [];
+
+    availableParams.forEach((param) => {
+      if (!Object.keys(req.body).includes(param)) {
+        errorArray.push(`${param} is required`);
+      }
+    });
+
+    if (!errorArray.length) {
+      return next();
+    }
+
+    const errorMessage = JSON.stringify(errorArray);
+    const error = new Error(errorMessage);
+    error.status = 400;
+    return next(error);
+  },
+  loginNonEmptyParamsValidator(req, res, next) {
+    const availableParams = ['usernameOrEmail', 'password'];
+    const errorArray = [];
+
+    availableParams.forEach((param) => {
+      if (!req.body[param].trim().length) {
+        errorArray.push(`${param} must not be empty`);
+      }
+    });
+
+    if (!errorArray.length) {
+      return next();
+    }
+
+    const errorMessage = JSON.stringify(errorArray);
+    const error = new Error(errorMessage);
+    error.status = 400;
+    return next(error);
+  },
+  async invalidCredential(req, res, next) {
+    const { usernameOrEmail, password } = req.body;
+    const loginUser = await User
+      .findOne({
+        where: {
+          [or]: [{ username: { [iLike]: usernameOrEmail } },
+            { email: { [iLike]: usernameOrEmail } }]
+        }
+      });
+
+    if (!loginUser) {
+      const error = new Error('Invalid Credential');
+      error.status = 400;
+      return next(error);
+    }
+    const valid = await loginUser.validPassword(password);
```

Comment:
Since the validPassword method on the user object returns a boolean, it may be best to make the variable name more inclined towards that. For example, `passwordIsValid`